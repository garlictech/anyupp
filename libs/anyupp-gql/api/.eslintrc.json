{
  "extends": "../../../.eslintrc.json",
  "ignorePatterns": [
    "!**/*"
  ],
  "rules": {},
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "plugins": [
        "@typescript-eslint"
      ],
      "parserOptions": {
        "project": [
          "libs/anyupp-gql/api/tsconfig.*?.json"
        ]
      }
    }
  ]
}
