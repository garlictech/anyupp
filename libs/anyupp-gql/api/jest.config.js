module.exports = {
  displayName: 'anyupp-gql-api',
  preset: '../../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/anyupp-gql/api',
};
