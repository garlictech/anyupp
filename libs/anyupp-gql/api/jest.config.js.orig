module.exports = {
  displayName: 'anyupp-gql-api',
  preset: '../../../jest.preset.js',
  globals: {
<<<<<<< HEAD
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
||||||| dbeda1c5
    'ts-jest': {
      tsConfig: '<rootDir>/tsconfig.spec.json',
    },
=======
    'ts-jest': { tsconfig: '<rootDir>/tsconfig.spec.json' },
>>>>>>> dev
  },
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/anyupp-gql/api',
};
