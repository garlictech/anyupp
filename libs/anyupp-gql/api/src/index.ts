export * from './lib/generated/appsync-api';
