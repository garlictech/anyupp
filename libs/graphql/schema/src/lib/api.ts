
/** ------------------------------------------------------
 * THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
 * -------------------------------------------------------
 */

/* tslint:disable */
/* eslint-disable */
export class UpdateAdminUserInput {
    email?: string;
    name?: string;
    phone?: string;
}

export class AdminUser {
    __typename?: 'AdminUser';
    email?: string;
    name?: string;
    phone?: string;
}

export abstract class IQuery {
    __typename?: 'IQuery';

    abstract hello(): string | Promise<string>;

    abstract getAdminUser(id: string): AdminUser | Promise<AdminUser>;
}

export abstract class ISubscription {
    __typename?: 'ISubscription';

    abstract adminUserChanged(id: string): AdminUser | Promise<AdminUser>;
}

export abstract class IMutation {
    __typename?: 'IMutation';

    abstract updateAdminUser(id: string, newAdminData: UpdateAdminUserInput): boolean | Promise<boolean>;
}
