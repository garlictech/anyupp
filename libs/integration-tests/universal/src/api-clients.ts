import {
  awsConfig,
  getCrudSdkForIAM,
  getCrudSdkForUserPool,
} from '@bgap/crud-gql/api';
import {
  getAnyuppSdkForIAM,
  getAnyuppSdkForUserPool,
} from '@bgap/anyupp-gql/api';
import { Auth } from 'aws-amplify';
import { from } from 'rxjs';
import { map } from 'rxjs/operators';

Auth.configure({
  ...awsConfig,
  // See: https://github.com/aws-amplify/amplify-js/issues/6552#issuecomment-682259256
  authenticationFlowType: 'USER_PASSWORD_AUTH',
  aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
});

export const createAuthenticatedCrudSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getCrudSdkForUserPool));

export const createAuthenticatedAnyuppSdk = (
  userName: string,
  password: string,
) => from(Auth.signIn(userName, password)).pipe(map(getAnyuppSdkForUserPool));

export const createIamCrudSdk = () =>
  getCrudSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );

export const createIamAnyuppSdk = () =>
  getAnyuppSdkForIAM(
    process.env.AWS_ACCESS_KEY_ID || '',
    process.env.AWS_SECRET_ACCESS_KEY || '',
  );
