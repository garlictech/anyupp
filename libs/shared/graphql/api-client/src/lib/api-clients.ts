import { awsConfig } from '@bgap/crud-gql/api';
import { config } from '@bgap/shared/config';
import { ICrudApiConfig } from '@bgap/shared/types';
import { GraphqlApiFp } from './graphql-api-fp';

const AWS_APPSYNC_CONFIG: ICrudApiConfig = {
  ...awsConfig,
  aws_appsync_apiKey: config.GraphqlApiKey,
  aws_appsync_graphqlEndpoint: config.GraphqlApiUrl,
};

export const appsyncGraphQlClient = GraphqlApiFp.createPublicClient(
  AWS_APPSYNC_CONFIG,
  console,
  true,
);

const AWS_AMPLIFY_CONFIG: ICrudApiConfig = {
  ...awsConfig,
};

export const amplifyGraphQlClient = GraphqlApiFp.createPublicClient(
  AWS_AMPLIFY_CONFIG,
  console,
  true,
);
