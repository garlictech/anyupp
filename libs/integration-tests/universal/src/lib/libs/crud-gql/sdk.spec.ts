import { CrudSdk, getCrudSdkForIAM } from '@bgap/crud-gql/api';
import {
  testAdminUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';
import { interval, of } from 'rxjs';
import {
  catchError,
  switchMap,
  switchMapTo,
  take,
  takeUntil,
  tap,
} from 'rxjs/operators';
import { createAuthenticatedCrudSdk } from '../../../api-clients';

describe('CRUD sdk test', () => {
  let sdk: CrudSdk;
  let authSdk: CrudSdk;
  const id = 'TEST_ADMIN_USER';

  beforeAll(async () => {
    const accessKeyId = process.env.AWS_ACCESS_KEY_ID || '';
    const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY || '';
    sdk = getCrudSdkForIAM(accessKeyId, secretAccessKey);
    authSdk = await createAuthenticatedCrudSdk(
      testAdminUsername,
      testAdminUserPassword,
    ).toPromise();
  });

  afterAll(async () => {
    await sdk.DeleteAdminUser({ input: { id } }).toPromise();
  });

  beforeEach(async () => {
    await sdk.DeleteAdminUser({ input: { id } }).toPromise();
  });

  test('An arbitrary CRUD', done => {
    const toMatchSnapshot = (x: any, name?: string) =>
      expect(x).toMatchSnapshot(
        {
          createdAt: expect.any(String),
          updatedAt: expect.any(String),
        },
        name,
      );

    sdk
      .DeleteAdminUser({ input: { id } })
      .pipe(
        switchMap(() => sdk.GetAdminUser({ id }, { fetchPolicy: 'no-cache' })),
        tap(x => expect(x).toMatchSnapshot('SHOULD BE NULL')),
        switchMap(() =>
          sdk.CreateAdminUser({
            input: {
              id,
              name: 'NAME',
              phone: '+6234567892',
              email: 'testuser+a1@anyupp.com',
            },
          }),
        ),
        tap(x => toMatchSnapshot(x, 'CREATE')),
        switchMap(() => sdk.GetAdminUser({ id }, { fetchPolicy: 'no-cache' })),
        tap(x => toMatchSnapshot(x, 'READ')),
        switchMap(() =>
          sdk.UpdateAdminUser({
            input: { id, name: 'UPDATED_USER' },
          }),
        ),
        tap(x => toMatchSnapshot(x, 'UPDATE')),
        switchMap(() => sdk.DeleteAdminUser({ input: { id } })),
        tap(x => expect(x).toMatchSnapshot('DELETE')),
        switchMap(() => sdk.GetAdminUser({ id }, { fetchPolicy: 'no-cache' })),
        tap(x =>
          expect(x).toMatchSnapshot('READ AFTER DELETE - SHOULD BE NULL'),
        ),
      )
      .subscribe(() => done());
  }, 60000);

  test('An arbitrary subscription', done => {
    const id = 'ADMIN_USERCRUD_SDK_ID';
    const dataSource$ = authSdk.OnAdminUserChange({ id });

    const subs$ = dataSource$.pipe(
      tap(x =>
        expect(x).toMatchSnapshot({
          createdAt: expect.any(String),
          updatedAt: expect.any(String),
        }),
      ),
      take(1),
    );

    const subsSubs = subs$.subscribe();

    authSdk
      .DeleteAdminUser({ input: { id } })
      .pipe(
        // we swallow the error if we cannot delete the user (probably it does not exist)
        catchError(() => of(true)),
        switchMap(() =>
          authSdk.CreateAdminUser({
            input: {
              id,
              name: 'NAME',
              phone: '+6234567890',
              email: 'testuser+a2@anyupp.com',
            },
          }),
        ),
        switchMapTo(interval(1000)),
        takeUntil(subs$),
        switchMap(() =>
          authSdk.UpdateAdminUser({
            input: { id, name: 'NAME2', email: 'a3@anyupp.com' },
          }),
        ),
      )
      .subscribe({
        complete: async () => {
          subsSubs.unsubscribe();
          await authSdk.DeleteAdminUser({ input: { id } }).toPromise();
          done();
        },
      });
  }, 60000);
});
