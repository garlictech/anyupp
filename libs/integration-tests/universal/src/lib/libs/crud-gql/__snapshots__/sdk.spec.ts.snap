// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUD sdk test An arbitrary CRUD: CREATE 1`] = `
Object {
  "createdAt": Any<String>,
  "email": "testuser+a1@anyupp.com",
  "id": "TEST_ADMIN_USER",
  "name": "NAME",
  "phone": "+6234567892",
  "profileImage": null,
  "roleContexts": Object {
    "items": Array [],
    "nextToken": null,
  },
  "settings": null,
  "updatedAt": Any<String>,
}
`;

exports[`CRUD sdk test An arbitrary CRUD: DELETE 1`] = `null`;

exports[`CRUD sdk test An arbitrary CRUD: READ 1`] = `
Object {
  "createdAt": Any<String>,
  "email": "testuser+a1@anyupp.com",
  "id": "TEST_ADMIN_USER",
  "name": "NAME",
  "phone": "+6234567892",
  "profileImage": null,
  "roleContexts": Object {
    "items": Array [],
    "nextToken": null,
  },
  "settings": null,
  "updatedAt": Any<String>,
}
`;

exports[`CRUD sdk test An arbitrary CRUD: READ AFTER DELETE - SHOULD BE NULL 1`] = `null`;

exports[`CRUD sdk test An arbitrary CRUD: SHOULD BE NULL 1`] = `null`;

exports[`CRUD sdk test An arbitrary CRUD: UPDATE 1`] = `
Object {
  "createdAt": Any<String>,
  "email": "testuser+a1@anyupp.com",
  "id": "TEST_ADMIN_USER",
  "name": "UPDATED_USER",
  "phone": "+6234567892",
  "profileImage": null,
  "roleContexts": Object {
    "items": Array [],
    "nextToken": null,
  },
  "settings": null,
  "updatedAt": Any<String>,
}
`;

exports[`CRUD sdk test An arbitrary subscription 1`] = `
Object {
  "createdAt": Any<String>,
  "email": "a3@anyupp.com",
  "id": "ADMIN_USERCRUD_SDK_ID",
  "name": "NAME2",
  "phone": "+6234567890",
  "profileImage": null,
  "roleContexts": Object {
    "items": Array [],
    "nextToken": null,
  },
  "settings": null,
  "updatedAt": Any<String>,
}
`;

exports[`CRUD sdk test An arbitrary subscription 2`] = `
Object {
  "createdAt": Any<String>,
  "email": "a3@anyupp.com",
  "id": "ADMIN_USERCRUD_SDK_ID",
  "name": "NAME2",
  "phone": "+6234567890",
  "profileImage": null,
  "roleContexts": Object {
    "items": Array [],
    "nextToken": null,
  },
  "settings": null,
  "updatedAt": Any<String>,
}
`;
